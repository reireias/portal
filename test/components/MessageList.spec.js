import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import tartget from '@/components/MessageList.vue'

const localVue = createLocalVue()

Vue.use(Vuetify)
localVue.use(Vuex)
localVue.use(Vuetify)

describe('MessageList', () => {
  let actions
  let getters
  let store
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
    actions = {
      bindMessages: jest.fn(),
    }
    getters = {
      messages: () => {
        return [{ text: 'dummy1' }, { text: 'dummy2' }, { text: 'dummy3' }]
      },
    }
    store = new Vuex.Store({
      state: {},
      getters,
      actions,
    })
  })

  test('is a Vue instance', () => {
    const wrapper = shallowMount(tartget, {
      localVue,
      store,
      vuetify,
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(actions.bindMessages).toHaveBeenCalled()
  })
})
