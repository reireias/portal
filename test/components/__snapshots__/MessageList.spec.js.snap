// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList is a Vue instance 1`] = `
"<v-container-stub tag=\\"div\\">
  <v-scroll-x-transition-stub group=\\"true\\" origin=\\"top center 0\\">
    <v-row-stub tag=\\"div\\" justify=\\"center\\" class=\\"message-row\\">
      <message-stub message=\\"[object Object]\\"></message-stub>
    </v-row-stub>
    <v-row-stub tag=\\"div\\" justify=\\"center\\" class=\\"message-row\\">
      <message-stub message=\\"[object Object]\\"></message-stub>
    </v-row-stub>
    <v-row-stub tag=\\"div\\" justify=\\"center\\" class=\\"message-row\\">
      <message-stub message=\\"[object Object]\\"></message-stub>
    </v-row-stub>
  </v-scroll-x-transition-stub>
</v-container-stub>"
`;
